src/pages/index.js

```diff
@@ -64,50 +129,107 @@ function handleImageClick(link, name) {
   popupWithImage.openModal(link, name);
 }
 
-// SECTION - GALLERY LIST -----
-
-const galleryListSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-
-      galleryListSection.addItem(cardElement);
-    },
-  },
-  ".gallery__list"
-);
-
-galleryListSection.renderItems();
-
 // ADD NEW CARD FORM -----
 
 const newCardFormPopup = new PopupWithForm({
   popupSelector: "#add-modal",
   handleFormSubmit: handleAddCardFormSubmit,
 });
 
-profileAddButton.addEventListener("click", () => newCardFormPopup.openModal());
+function handleDeleteClick(cardData) {
+  confirmPopup.openModal(cardData);
+}
+
+profileAddButton.addEventListener("click", () => {
+  cardFormValidator.resetValidation();
+  newCardFormPopup.openModal();
+});
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.generateCard();
   return cardElement;
 }
 
+function handleLikeClick(card) {
+  const cardId = card.getId();
+  if (card.getLikeStatus()) {
+    api
+      .unlikeCard(cardId)
+      .then(() => {
+        card.setLikeStatus(false);
+      })
+      .catch((err) => console.error(`${err} Something went wrong`));
+  } else {
+    api
+      .likeCard(cardId)
+      .then(() => {
+        card.setLikeStatus(true);
+      })
+      .catch((err) => console.error(`${err} Something went wrong`));
+  }
+}
+
 newCardFormPopup.setEventListeners();
 
 function handleAddCardFormSubmit(inputValues) {
-  const cardElement = createCard(inputValues);
-  galleryListSection.addItem(cardElement);
-  newCardFormPopup.closeModal();
-  cardFormValidator.toggleButtonState();
+  newCardFormPopup.setButtonText("Saving...");
+  api
+    .addNewCard(inputValues.name, inputValues.link)
+    .then((data) => {
+      const cardElement = createCard(data);
+      galleryListSection.addItem(cardElement);
+      newCardFormPopup.closeModal();
+      cardFormValidator.toggleButtonState();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      newCardFormPopup.setButtonText("Create");
+    });
+}
+
+//CONFIRMATION POPUP -----
+
+function handleSubmit(card) {
+  const cardId = card.getId();
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      card.removeCard();
+    })
+    .catch((err) => console.error(err));
+  confirmPopup.closeModal();
```

Comment:
`  confirmPopup.closeModal();` should be in line 204


You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well